extensions/strategies/trendline/strategy.js

```diff
@@ -1,113 +1,66 @@
-      for (let i = 0; i < s.options.avgpoints; i++) { tll.push(s.lookback[i].close) }
 
       var chart = tl1
 
       var growth = trend(chart, {
         lastPoints: s.options.lastpoints,
         avgPoints: s.options.avgpoints,
         avgMinimum: 0,
-        reversed: true
+        reversed: false
       })
-      var growth2 = trend(chart, {
-        lastPoints: s.options.lastpoints2,
-        avgPoints: s.options.avgpoints2,
-        avgMinimum: 0,
-        reversed: true
-      })
-
-      s.stats = growth
-      s.growth = growth > 1
-      s.stats2 = growth2
-      s.growth2 = growth2 > 1
-      s.stdevs = stats.stdev(tls)
-      s.stdevl = stats.stdev(tll)
-      s.means = math.mean(tls)
-      s.meanl = math.mean(tll)
-      s.pcts = s.stdevs / s.means
-      s.pctl = s.stdevl / s.meanl
-      s.options.markup_sell_pct = math.mean(s.pcts, s.pctl) * 100
-      s.options.markdown_buy_pct = math.mean(s.pcts, s.pctl) * 100
-      s.accel = growth > oldgrowth
-      oldgrowth = growth
+    
+      global.direc = growth > s.options.trend
+      global.gradient = growth
     }
-
-    if (
-      s.growth === true &&
-         s.growth2 === true
-    )
-    {
-      s.signal = 'buy'
+    if (global.direc === false) {
+      s.signal = 'sell'
     }
-    else if (
-      s.growth === false |
-         s.growth2 === false |
-         s.accel === false
-    )
-    {
-      //s.signal = 'sell'
+    else if (global.direc === true) {
+      s.signal = 'buy'
     }
+	global.oldgrowth = global.gradient
     cb()
   },
   onReport: function (s) {
     var cols = []
-    cols.push(' ')
-    cols.push(z(8, n(s.stats).format('0.00000000'), ' ')[s.growth === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.stats2).format('0.00000000'), ' ')[s.growth2 === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.stdevs).format('0.00000000'), ' ')[s.accel === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.stdevl).format('0.00000000'), ' ')[s.accel === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.means).format('0.00000000'), ' ')[s.accel === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.meanl).format('0.00000000'), ' ')[s.accel === true ? 'green' : 'red'])
-    cols.push(' ')
-    cols.push(z(8, n(s.options.markup_sell_pct).format('0.00000000'), ' ')[s.accel === true ? 'green' : 'red'])
+    cols.push(z(8, n(global.gradient).format('0.00000000'), ' ')[global.gradient === true ? 'green' : 'red'])
     return cols
   },
 
   phenotypes: {
     // -- common
-    period_length: Phenotypes.RangePeriod(1, 400, 'm'),
-    min_periods: Phenotypes.Range(1, 200),
+    // reference in extensions is given in ms have not heard of an exchange that supports 500ms thru api so setting min at 1 second
+    period_length: Phenotypes.RangePeriod(1, 7200, 's'),
+    min_periods: Phenotypes.Range(1, 2500),
```

Comment:
not sure there how to solve it. you need to have min_periods > avgpoints and min_periods lastpoints, otherwise you will get a TypeError. I will predict there will be issues created.